uniffi_bindgen/src/interface/mod.rs

```diff
@@ -100,30 +100,69 @@ impl<'ci> ComponentInterface {
         Ok(ci)
     }
 
+    /// The string namespace within which this API should be presented to the caller.
+    ///
+    /// This string would typically be used to prefix function names in the FFI, to build
+    /// a package or module name for the foreign language, etc.
     pub fn namespace(&self) -> &str {
         self.namespace.as_str()
     }
 
+    /// List the definitions for every Enum type in the interface.
     pub fn iter_enum_definitions(&self) -> Vec<Enum> {
         self.enums.to_vec()
     }
 
+    /// Get an Enum definition by name, or None if no such Enum is defined.
+    pub fn get_enum_definition(&self, name: &str) -> Option<&Enum> {
+        // TODO: probably we could store these internally in a HashMap to make this easier?
+        self.enums.iter().find(|e| e.name == name)
+    }
+
+    /// List the definitions for every Record type in the interface.
     pub fn iter_record_definitions(&self) -> Vec<Record> {
         self.records.to_vec()
     }
 
+    /// Get a Record definition by name, or None if no such Record is defined.
+    pub fn get_record_definition(&self, name: &str) -> Option<&Record> {
+        // TODO: probably we could store these internally in a HashMap to make this easier?
+        self.records.iter().find(|r| r.name == name)
+    }
```

Comment:
IIUC, we only need this method, but not the other `get_*_definition` methods. I don't understand how this wasn't needed in the other bindings.

Reply:
Right, we only need `get_record_definition` but I added the others for consistency.

The reason they'd needed here is because the python bindings generate some code for `type::Record` instances as part of doing `ci.iter_types`, while e.g. the Kotlin bindings generate all the code for `type::Record` while doing `ci.iter_record_definitions`. For python, we need to be able to go from the `type::Record(name)` that we get when iterating the types, to the actual definition of the record with the given name.